File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.java
class Solution {
    public int minOperations(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap();
        for(int x : nums) mp.put(x, mp.getOrDefault(x, 0)+1);
        int ans = 0;
        for(int cnt : mp.values()){
            if(cnt == 1) return -1;
            ans += Math.ceil((double)cnt/3);
        }
        return ans;
    }
}

File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        unordered_map<int, int> cnt;
        for(int x : nums) cnt[x]++;
        int ans = 0;
        for(auto [_, val] : cnt){
            if(val==1) return -1;
            ans += ceil((double)(val)/3);
        }
        return ans;
    }
};